bot/exts/utilities/color.py

```diff
@@ -0,0 +1,330 @@
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @commands.command(aliases=["colour"])
+    async def color(self, ctx: commands.Context, mode: str, *, user_color: str) -> None:
+        """
+        Send information on input color code or color name.
+
+        Possible modes are: "hex", "rgb", "hsv", "hsl", "cmyk" or "name".
+        """
+        logger.debug(f"{mode = }")
+        logger.debug(f"{user_color = }")
+        color_name = None
+        if mode.lower() == "hex":
+            hex_match = re.fullmatch(r"(#?[0x]?)((?:[0-9a-fA-F]{3}){1,2})", user_color)
+            if hex_match:
+                hex_color = int(hex(int(user_color.replace("#", ""), 16)), 0)
+                if "#" in user_color:
+                    rgb_color = ImageColor.getcolor(user_color, "RGB")
+                elif "0x" in user_color:
+                    hex_ = user_color.replace("0x", "#")
+                    rgb_color = ImageColor.getcolor(hex_, "RGB")
+                else:
+                    hex_ = "#" + user_color
+                    rgb_color = ImageColor.getcolor(hex_, "RGB")
+            else:
+                await ctx.send(
+                    embed=Embed(
+                        title="There was an issue converting the hex color code.",
+                        description=ERROR_MSG.format(user_color=user_color),
+                    )
+                )
+        elif mode.lower() == "rgb":
+            rgb_color = self.tuple_create(user_color)
+        elif mode.lower() == "hsv":
+            hsv_temp = self.tuple_create(user_color)
+            rgb_color = self.hsv_to_rgb(hsv_temp)
+        elif mode.lower() == "hsl":
+            hsl_temp = self.tuple_create(user_color)
+            rgb_color = self.hsl_to_rgb(hsl_temp)
+        elif mode.lower() == "cmyk":
+            cmyk_temp = self.tuple_create(user_color)
+            rgb_color = self.cmyk_to_rgb(cmyk_temp)
+        elif mode.lower() == "name":
+            color_name, hex_color = self.match_color_name(user_color)
+            if "#" in hex_color:
+                rgb_color = ImageColor.getcolor(hex_color, "RGB")
+            else:
+                rgb_color = ImageColor.getcolor("#" + hex_color, "RGB")
+        else:
+            # mode is either None or an invalid code
+            if mode is None:
+                no_mode_embed = Embed(
+                    title="No 'mode' was passed, please define a color code.",
+                    description="Possible modes are: Name, Hex, RGB, HSV, HSL and CMYK.",
+                    color=Colours.soft_red,
+                )
+                await ctx.send(embed=no_mode_embed)
+                return
+            wrong_mode_embed = Embed(
+                title=f"The color code {mode} is not a valid option",
+                description="Possible modes are: Name, Hex, RGB, HSV, HSL and CMYK.",
+                color=Colours.soft_red,
+            )
+            await ctx.send(embed=wrong_mode_embed)
+            return
```

Comment:
I would prefer if these were subcommands that called a main method which would send the embed.

Reply:
The recent commits should have addressed that as I understand it, I went ahead and restructured the script with subcommands. With all of the modes and conversions going on it gets a bit lengthy. [commit link](https://github.com/python-discord/sir-lancebot/pull/842/commits/4f52cad537e19b3313b726d099ac223a3fa31c5c)